contracts/axone-cognitarium/src/contract.rs

```diff
@@ -36,15 +36,19 @@ pub fn execute(
     info: MessageInfo,
     msg: ExecuteMsg,
 ) -> Result<Response, ContractError> {
-    match msg {
-        ExecuteMsg::InsertData { format, data } => {
-            execute::insert(deps, info, format.unwrap_or_default(), data)
+    if !info.funds.is_empty() {
+        Err(ContractError::NoFundAccepted)
+    } else {
+        match msg {
+            ExecuteMsg::InsertData { format, data } => {
+                execute::insert(deps, info, format.unwrap_or_default(), data)
+            }
+            ExecuteMsg::DeleteData {
+                prefixes,
+                delete,
+                r#where,
+            } => execute::delete(deps, info, prefixes, delete, r#where),
```

Comment:
The code is correct, but I have additional requirements to specify (sorry 😛 ). Since these are pre-conditions that need to be verified, as a best practice, I'd say it’s important not to chain conditional blocks and to exit as early as possible to clearly separate what pertains to checks from the functional aspects. Furthermore, I think it would be even more explicit to use the [`ensure!`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/macro.ensure.html) macro provided by [`cosmwasm_std`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/) to capture the pre-conditions. For instance:

```rust
ensure!(info.funds.is_empty(),  ContractError::NoFundAccepted);
```

I find this macro simplifies the expression of preconditions by clearly delimiting the condition. Additionally, by highlighting the expected state rather than the incorrect state, it makes the code clearer and more readable, allowing the requirements to be understood at a glance, which is imho.

What do you think?

Reply:
I'm completely agree ! I didn't know about this [`ensure!`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/macro.ensure.html) macro, it's better and more readable like you say, I will made the changes, thanks!